Azure/azure-cli
# [AppConfig] Support AAD auth for data operations

**Description**  

App Configuration supports AAD authentication and authorization for all data operations. Since we are now using the [official Python SDK for AppConfiguration](https://github.com/Azure/azure-sdk-for-python/tree/master/sdk/appconfiguration/azure-appconfiguration), we can easily support AAD auth in appconfig CLI module by utilizing the `az login` credentials.
This PR introduces two new arguments for users to authenticate using AAD credentials:

- --auth-mode (login or key; default is key which preserves the current behavior)

- --endpoint (AppConfiguration endpoint like `https://storename.azconfig.io`)

For any existing AppConfig commands, providing the `--name` and `--connection-string` parameters will use key (HMAC) auth for data operations (preserving current behavior). Instead of these two arguments, if users provide `--auth-mode login` and `--endpoint` parameters, AAD credentials will be used to connect to AzureAppConfigurationClient.

**Testing Guide**  


1. Create a new App Configuration with name "aadtestcli".
2. Create a key-value with key "testKey" and any value.
3. Create a new service principal and assign it "App Configuration Data Reader" role in your App Configuration. Wait for about 15 minutes for permissions to propagate.
4. Login to CLI with the service principal credentials.


This read request should succeed:
`az appconfig kv show --key testKey --auth-mode login --endpoint https://aadtestcli.azconfig.io`

This write request should fail with 'Forbidden' error:
`az appconfig kv lock --key testKey --auth-mode login --endpoint https://aadtestcli.azconfig.io`


**History Notes**  



---

This checklist is used to make sure that common guidelines for a pull request are followed.
